### PR TITLE
Reduce overhead in SubsetSelectorImpl by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead in SubsetSelectorImpl by fixing alignment
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/load_balancer_impl.h [hunk 1]
```diff
@@ -757,10 +757,10 @@ class SubsetSelectorImpl : public SubsetSelector {
 
 private:
   const std::set<std::string> selector_keys_;
+  const std::set<std::string> fallback_keys_subset_;
```
Comment (adisuissa) [inline source/common/upstream/load_balancer_impl.h:760]: Why was this field moved to the end?

**Review by adisuissa — COMMENTED** [review]
Thanks! Left a question.
/wait-any

File: source/common/upstream/load_balancer_impl.h (hunk 1 above)
Comment (DiazAlan) [reply source/common/upstream/load_balancer_impl.h:760]: I wasn't moved down, it was moved up above an enum

Comment (adisuissa) [reply source/common/upstream/load_balancer_impl.h:760]: Ah... misread the change. That makes sense.
Can you please add a relevant comment (before the enum) similar to other PRs?

Comment (DiazAlan) [reply source/common/upstream/load_balancer_impl.h:760]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Pull request merged**